jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -167,6 +167,13 @@ static void afterClinitInitialization() {
 			unsafe.putObject(unsafe.staticFieldBase(f2), unsafe.staticFieldOffset(f2), com.ibm.jit.JITHelpers.getHelpers());
 		} catch (NoSuchFieldException e) { }
 		
+		/**
+		 * When the System Property == true, then enable copying in String.substring(int) and String.substring(int, int)
+		 * methods whenever offset is zero. Otherwise, disable copy.
+		 */
+		String enableCopyInSubstringWhenOffsetIsZeroProperty = internalGetProperties().getProperty("java.lang.string.substring.copy.offset.zero"); //$NON-NLS-1$
+		String.enableCopyInSubstringWhenOffsetIsZero = enableCopyInSubstringWhenOffsetIsZeroProperty != null && !enableCopyInSubstringWhenOffsetIsZeroProperty.equalsIgnoreCase("false"); //$NON-NLS-1$
```

Comment:
This becomes
```
enableSharingInSubstringWhenOffsetIsZero = enableCopyInSubstringWhenOffsetIsZeroProperty == null || enableCopyInSubstringWhenOffsetIsZeroProperty.equalsIgnoreCase("false");
```

Reply:
@keithc-ca your suggestion isn't quite the same. enableCopyInSubstringWhenOffsetIsZero should only be set to true when there is a system property set to some value other than false.

Reply:
Ah, nm, I see this comment is a continuation of the earlier comment.

Reply:
Fixed in a44d52e. Though it should be `...equalsIgnoreCase("true")`, right?

Reply:
@pshipton this is the intent though. `String.enableSharingInSubstringWhenOffsetIsZero` should be `true` by default. The system property is only provided as a fallback in case users experience OOM issues in very rare cases outlined in the commit description.